### PR TITLE
OGD-430 implemented live paye endpoint

### PR DESCRIPTION
**Pull request by edinhodzic**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/individualsemploymentsapi/service/LiveEmploymentsServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  private val liveEmploymentsService = new LiveEmploymentsService(individualsMatchingApiConnector, desConnector)
+
+  private val randomMatchId = UUID.randomUUID()
+  private val nino = Nino("AB123456C")
+
+  implicit val hc = new HeaderCarrier
+
+  "Live employments service paye function" should {
+
+    "return employments for the entire available history ordered by date descending" ignore {
```
Comment (edgarjimenez) [inline test/unit/uk/gov/hmrc/individualsemploymentsapi/service/LiveEmploymentsServiceSpec.scala]: a bunch of ignored tests here

File: app/uk/gov/hmrc/individualsemploymentsapi/connector/DesConnector.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.hmrc.individualsemploymentsapi.util.JsonFormatters._
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.logging.Authorization
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpGet, HttpReads, NotFoundException}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class DesConnector @Inject()(configuration: Configuration) extends ServicesConfig {
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/individualsemploymentsapi/connector/DesConnector.scala]: Can't find any tests for this connector

File: app/uk/gov/hmrc/individualsemploymentsapi/domain/Employment.scala [hunk 3]
```diff
@@ -18,5 +18,29 @@ package uk.gov.hmrc.individualsemploymentsapi.domain
 
 import org.joda.time.LocalDate
 import uk.gov.hmrc.individualsemploymentsapi.domain.PayFrequency.PayFrequency
+import uk.gov.hmrc.individualsemploymentsapi.domain.des.{DesAddress, DesEmployment}
 
 case class Employment(startDate: Option[LocalDate], endDate: Option[LocalDate], employer: Option[Employer], payFrequency: Option[PayFrequency])
+
+object Employment {
+
+  def from(desEmployment: DesEmployment): Option[Employment] = {
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/individualsemploymentsapi/domain/Employment.scala]: all this needs testing

File: test/unit/uk/gov/hmrc/individualsemploymentsapi/domain/EmploymentSpec.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import uk.gov.hmrc.individualsemploymentsapi.domain.{Address, Employer, Employment, PayFrequency}
+
+class EmploymentSpec extends FlatSpec with Matchers {
+
+  "Employment" should "derive itself from an instance of des employment" in {
+    val desEmployment = DesEmployment(Seq.empty, Some("Acme Inc"), Some(DesAddress("line 1", "AB1 2CD")), Some("123"), Some("AB12345"), Some(parse("2016-01-01")), Some(parse("2016-12-31")), Some(DesPayFrequency.M1))
+    val employment = Employment(Some(parse("2016-01-01")), Some(parse("2016-12-31")), Some(Employer(Some(EmpRef("123", "AB12345")), Some("Acme Inc"), Some(Address("line 1", None, None, None, None, "AB1 2CD")))), Some(PayFrequency.withName("CALENDAR_MONTHLY")))
+    Employment.from(desEmployment) shouldBe Some(employment)
+  }
+
```
Comment (edgarjimenez) [inline test/unit/uk/gov/hmrc/individualsemploymentsapi/domain/EmploymentSpec.scala]: we should also test that an employment is None when all of its fields are None.

File: app/uk/gov/hmrc/individualsemploymentsapi/domain/Employer.scala [hunk 5]
```diff
@@ -19,3 +19,13 @@ package uk.gov.hmrc.individualsemploymentsapi.domain
 import uk.gov.hmrc.domain.EmpRef
 
 case class Employer(payeReference: Option[EmpRef], name: Option[String], address: Option[Address])
+
+object Employer {
+  def create(payeReference: Option[EmpRef], name: Option[String], address: Option[Address]): Option[Employer] = {
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/individualsemploymentsapi/domain/Employer.scala]: we should have tests for this, just as we have for Employment
